packages/CodingStandard/src/Fixer/Property/ArrayPropertyDefaultValueFixer.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\Property;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+final class ArrayPropertyDefaultValueFixer extends AbstractFixer
+{
+    public function isCandidate(Tokens $tokens): bool
+    {
+        // analyze only properties with comments
+        return $tokens->isAllTokenKindsFound([T_DOC_COMMENT, T_VARIABLE]);
+    }
+
+    protected function applyFix(SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($tokens as $index => $token) {
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            if (! $this->isArrayPropertyDocComment($token)) {
+                continue;
+            }
+
+            $semicolonSignOrArrayOpenerPosition = $tokens->getNextMeaningfulToken($index + 4);
+            $semicolonSignOrArrayOpenerToken = $tokens[$semicolonSignOrArrayOpenerPosition];
+            if ($semicolonSignOrArrayOpenerToken->equals('[')) {
+                // token after property equal is start of an array
+                continue;
+            }
+
+            // token after property is ; - its end => so no definition
+            $this->addDefaultValueForArrayProperty($semicolonSignOrArrayOpenerToken);
+        }
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Array property should have default value, to prevent undefined array issues.',
+            []
+        );
+    }
+
+    private function isArrayPropertyDocComment(Token $token): bool
+    {
+        $docBlock = new DocBlock($token->getContent());
+
+        if (count($docBlock->getLines()) === 1) {
+            return false;
+        }
+
+        if (! $docBlock->getAnnotationsOfType('var')) {
+            return false;
+        }
+
+        $varAnnotation = $docBlock->getAnnotationsOfType('var')[0];
+        if (! Strings::contains($varAnnotation->getTypes()[0], '[]')) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private function addDefaultValueForArrayProperty(Token $semicolonToken): void
+    {
+        $semicolonToken->setContent(' = [];');
```

Comment:
You can do a few things here;
- remove the token and insert new ones
- override the token and after that insert new ones (or in other order)
- override the token and insert new ones (in one go)

It doesn't matter which one you pick (not much at least), the important part is that each token is added to the Tokens collection and the one you no longer want is removed.

In this case I would do something like;
```php
    private function addDefaultValueForArrayProperty(Tokens $tokens, int $semicolonTokenIndex): void
    {
        $tokens[$semicolonTokenIndex]->clear();
        $tokens->insertAt($semicolonTokenIndex, new Token([T_WHITESPACE, ' ']));
        $tokens->insertAt($semicolonTokenIndex, new Token('='));
        $tokens->insertAt($semicolonTokenIndex, new Token([T_WHITESPACE, ' ']));
        $tokens->insertAt($semicolonTokenIndex, new Token([CT::T_ARRAY_SQUARE_BRACE_OPEN, '[']));
        $tokens->insertAt($semicolonTokenIndex, new Token([CT::T_ARRAY_SQUARE_BRACE_CLOSE, ']']));
    }
```

note: `CT::` is in namespace namespace PhpCsFixer\Tokenizer
note 2: didn't test the code ;)